src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -2079,14 +2102,16 @@ public void run() {
                 // Follow refresh logic now. Authority is valid or
                 // skipped validation
                 final AuthenticationResult authResult;
-                try
-                {
+                try {
                     authResult = getTokenWithRefreshToken(null, false, request, refreshItem, false);
-                }
-                catch (final AuthenticationException authException)
-                {
+                } catch (final AuthenticationException authException) {
                     callbackHandle.onError(authException);
                     return;
+                } catch (final HttpResponseException httpResponseException) {
+                    final AuthenticationException authenticationException = new AuthenticationException(
+                            ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, httpResponseException.getMessage());
+                    callbackHandle.onError(authenticationException);
```

Comment:
Then this change is also changing behavior, before this change for IOexceptions (occurring in postMessage) in response we were returning AUTH_FAILED_NO_TOKEN , now we return AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, because we will throw HttpResponseException.
So why this is allowed and another way around is not?